public/js/clients/firefox/events.js

```diff
@@ -59,19 +65,18 @@ function resumed(_, packet) {
 function newSource(_, packet) {
   const { source } = packet;
 
-  if (NEW_SOURCE_IGNORED_URLS.indexOf(source.url) > -1) {
-    return;
-  }
-
-  if (source.introductionType == "debugger eval") {
-    return;
+  source.hidden = false;
+  if (isHidden(source)) {
+    source.url = `evals/console-evauation-${evalIndex++}.js`;
+    source.hidden = true;
```

Comment:
Some clarification: we need to generate URLs for these types of sources anyway (eval-ed, original) so we might as well encode that information in there.